apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/PathExpectationSamples.kt

```diff
@@ -29,12 +28,362 @@ class PathExpectationSamples {
+
+        expect(dir).toBeExecutable()
+
+        fails {
+            expect(Paths.get(INVALID_DIR)).toBeExecutable()
+        }
+    }
+
+    @Test
+    fun toBeARegularFile() {
+        val file = tempDir.newFile(TEST_FILE)
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(file).toBeARegularFile()
+
+        fails {
+            expect(dir).toBeARegularFile()
+        }
+    }
+
+
+    @Test
+    fun toBeADirectory() {
+        val file = tempDir.newFile(TEST_FILE)
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).toBeADirectory()
+
+        fails {
+            expect(file).toBeADirectory()
+        }
+    }
+
+    @Test
+    fun toBeAbsolute() {
+        val dir = tempDir.newDirectory(ABSOLUTE_DIR)
+
+        expect(dir).toBeAbsolute()
+
+        fails {
+            expect(Paths.get(RELATIVE_DIR)).toBeAbsolute()
+        }
+    }
+
+    @Test
+    fun toBeRelative() {
+        expect(Paths.get(RELATIVE_DIR)).toBeRelative()
+
+        fails {
+            expect(tempDir.newDirectory(ABSOLUTE_DIR)).toBeRelative()
+        }
+    }
+
+    @Test
+    fun toHaveTheDirectoryEntries() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+        val file = dir.newFile(TEST_FILE)
+
+        expect(dir).toHaveTheDirectoryEntries(TEST_FILE)
+
+        file.delete()
+
+        fails {
+            expect(dir).toHaveTheDirectoryEntries(TEST_FILE)
+        }
+
+    }
+
+    @Test
+    fun toHaveTheSameTextualContentAs() {
+        val writtenTextInFile = "test_test"
+
+        val notEmptyFilePath = tempDir.newFile("${TEST_FILE}_1")
+        notEmptyFilePath.toFile().writeText(writtenTextInFile)
+
+        val expectedFilePath = tempDir.newFile("${TEST_FILE}_2")
+        expectedFilePath.toFile().writeText(writtenTextInFile)
+
+        val emptyFilePath = tempDir.newFile("${TEST_FILE}_3")
+
+        expect(notEmptyFilePath).toHaveTheSameTextualContentAs(expectedFilePath)
+
+        fails {
+            expect(emptyFilePath).toHaveTheSameTextualContentAs(expectedFilePath)
+        }
+    }
+
+    @Test
+    fun toHaveTheSameBinaryContentAs() {
+        val notEmptyFilePath = tempDir.newFile("${TEST_FILE}_1")
+        notEmptyFilePath.toFile().writeBytes(byteArrayOf(1, 2, 3))
+
+        val expectedFilePath = tempDir.newFile("${TEST_FILE}_2")
+        expectedFilePath.toFile().writeBytes(byteArrayOf(1, 2, 3))
+
+        val emptyFilePath = tempDir.newFile("${TEST_FILE}_3")
+
+        expect(notEmptyFilePath).toHaveTheSameBinaryContentAs(expectedFilePath)
+
+        fails {
+            expect(emptyFilePath).toHaveTheSameBinaryContentAs(expectedFilePath)
+        }
+    }
+
+    @Test
+    fun testPathExtensionFeature() {
+        val extension = "txt"
+        val dir = tempDir.newFile("$TEST_FILE.$extension")
+
+        expect(dir).extension.notToBeEmpty().toEqual(extension)
+
+        fails {
+            val invalidExtension = "txtt"
+            expect(dir).extension.toEqual(invalidExtension)
```

Comment:
Please have a look at e.g. ListExpectationSamples -> getFeature. It would be nice if you could include the comment about the fail fast behaviour and about the subject change, something like
```suggestion
            expect(dir)
              .extension         // subject is now of type String (actually "txt")
              .toStartWith("jp") // fails because extension starts with `tx`
              .toEndWith("eg")   // not reported
              //                    use `extension { ... }` if you want that all expectations are evaluated
```